src/Surfnet/StepupRa/RaBundle/Security/Authentication/Token/SamlToken.php

```diff
@@ -39,18 +39,19 @@ class SamlToken extends AbstractToken
      */
     private $institutionConfigurationOptions;
 
-    public function __construct(
-        Loa $loa,
-        array $roles = [],
-        InstitutionConfigurationOptions $institutionConfigurationOptions = null
-    ) {
+    public function __construct(Loa $loa, array $roles = [])
```

Comment:
Thanks for the comments. I agree with your proposal and would like to implement it soon (and add some tests while at it).This eliminates the setter which is a good thing and adds an important guard.

However, I'm not sure how much this cleans up the controller and you would still need to get the token, token storage and options, modify the token and save it. Or there might be something I'm missing.

A command handler would indeed be very nice for cleaning up the controller(s), although this feels more like a tech debt issue.